extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SkipPredicateInvoker.java

```diff
@@ -13,37 +16,32 @@
  *
  * @see Scheduled#skipExecutionIf()
  */
-public final class SkipPredicateInvoker implements ScheduledInvoker {
+public final class SkipPredicateInvoker extends DelegateInvoker {
 
-    private static final Logger LOGGER = Logger.getLogger(SkipPredicateInvoker.class);
+    private static final Logger LOG = Logger.getLogger(SkipPredicateInvoker.class);
 
-    private final ScheduledInvoker delegate;
     private final Scheduled.SkipPredicate predicate;
     private final Event<SkippedExecution> event;
 
     public SkipPredicateInvoker(ScheduledInvoker delegate, Scheduled.SkipPredicate predicate,
             Event<SkippedExecution> event) {
-        this.delegate = delegate;
+        super(delegate);
         this.predicate = predicate;
         this.event = event;
     }
 
     @Override
-    public void invoke(ScheduledExecution execution) throws Exception {
+    public CompletionStage<Void> invoke(ScheduledExecution execution) throws Exception {
         if (predicate.test(execution)) {
-            LOGGER.debugf("Skipped scheduled invoker execution: %s", delegate.getClass().getName());
+            LOG.debugf("Skipped scheduled invoker execution: %s", delegate.getClass().getName());
             SkippedExecution payload = new SkippedExecution(execution,
                     predicate.getClass().getName());
             event.fire(payload);
             event.fireAsync(payload);
```

Comment:
Same comment here, should we wait for the async observer to complete before returning the completion stage?